src/tiled/editpolygontool.cpp

```diff
@@ -566,6 +583,91 @@ void EditPolygonTool::updateMovingItems(const QPointF &pos,
     }
 }
 
+void EditPolygonTool::addNode(const QPointF &pos,
+                              Qt::KeyboardModifiers modifiers)
+{
+    MapRenderer *renderer = mapDocument()->renderer();
+    QPointF pixelCoords = renderer->screenToPixelCoords(pos);
+
+    SnapHelper(renderer, modifiers).snap(pixelCoords);
+
+    const auto &selectedHandle = *mSelectedHandles.begin();
+    const MapObjectItem *item = selectedHandle->mapObjectItem();
+    MapObject *mapObject = item->mapObject();
+
+    pixelCoords -= mapObject->position();
+
+    QPolygonF polygon = mapObject->polygon();
+
+    if ((polygon.last() - polygon.first()).manhattanLength() < QApplication::startDragDistance() * 1.0)
```

Comment:
Maybe introduce a variable for the left side of the `<` here, to keep the line shorter?